bioptim/limits/penalty.py

```diff
@@ -346,6 +346,53 @@ def minimize_predicted_com_height(penalty, ocp, nlp, t, x, u, p):
                 CoM_height = (CoM_dot[2] * CoM_dot[2]) / (2 * -g) + CoM[2]
                 penalty.type.get_type().add_to_penalty(ocp, nlp, CoM_height, penalty)
 
+        @staticmethod
+        def minimize_com_height(penalty, ocp, nlp, t, x, u, p):
+            """
+            Adds the objective that the height of the center of mass of the model should be minimized.
+            The height is assumed to be the third axis.
+            """
+
+            target = None
+            if penalty.target is not None:
+                target = PenaltyFunctionAbstract._check_and_fill_tracking_data_size(penalty.target, (1, len(x)))
+
+            PenaltyFunctionAbstract._add_to_casadi_func(nlp, "biorbd_CoM", nlp.model.CoM, nlp.q)
+            for i, v in enumerate(x):
+                q = nlp.mapping["q"].expand.map(v[: nlp.shape["q"]])
+                CoM = nlp.casadi_func["biorbd_CoM"](q)
+                CoM_height = CoM[2]
+                penalty.sliced_target = target[:, i] if target is not None else None
+                penalty.type.get_type().add_to_penalty(ocp, nlp, CoM_height, penalty)
+
+        @staticmethod
+        def minimize_com_velocity(penalty, ocp, nlp, t, x, u, p, axis=None):
+            """
+            Adds the objective that the velocity of the center of mass of the model should be minimized.
+            If no axis is specified, the squared-norm of the CoM's velocity is mimimized.
+            Otherwise, the projection of the CoM's velocity on the specified axis is minimized.
+            """
+
+            target = None
+            if penalty.target is not None:
+                target = PenaltyFunctionAbstract._check_and_fill_tracking_data_size(penalty.target, (1, len(x)))
+
+            PenaltyFunctionAbstract._add_to_casadi_func(nlp, "biorbd_CoMdot", nlp.model.CoMdot, nlp.q, nlp.q_dot)
+            for i, v in enumerate(x):
+                q = nlp.mapping["q"].expand.map(v[: nlp.shape["q"]])
+                q_dot = nlp.mapping["q_dot"].expand.map(v[nlp.shape["q"] :])
+                CoM_dot = nlp.casadi_func["biorbd_CoMdot"](q, q_dot)
+
+                if axis == None:
+                    CoM_dot_proj = CoM_dot[0] ** 2 + CoM_dot[1] ** 2 + CoM_dot[2] ** 2
+                elif not isinstance(axis, Axe):
+                    raise RuntimeError("axis must be a bioptim.Axe")
+                else:
+                    CoM_dot_proj = CoM_dot[axis]
```

Comment:
Indeed, since the logic is there, I accept your suggestion to call `minimize_com_height` `minimize_com_position` and add axis.